cpp/inc/bond/ext/grpc/unary_call.h

```diff
@@ -87,163 +134,282 @@ namespace detail {
             bool wasResponseSent = _responseSentFlag.test_and_set();
             if (!wasResponseSent)
             {
-                _self = this->shared_from_this();
                 _responder.FinishWithError(status, static_cast<void*>(this));
             }
         }
 
         void invoke(bool /* ok */) override
         {
-            // response has been sent, so we no longer need to keep
-            // ourselves alive
-            _self.reset();
+            // The response has been sent, so we no longer need to keep
+            // ourselves alive: release the implicit initial refcount that
+            // this instance was constructed with.
+            Release(ReleaseTrySend::No);
         }
     };
 
-}
+    template <typename TRequest, typename TResponse>
+    void intrusive_ptr_add_ref(unary_call_impl<TRequest, TResponse>* call)
```

Comment:
Both `intrusive_ptr_add_ref` and `intrusive_ptr_release` should become `friend`s?

Reply:
And, if not, they need to be inline.

Reply:
The in-class defined functions are implicitly inline.